source/GCS_util.go

```diff
@@ -0,0 +1,95 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/conduitio/conduit-connector-google-cloudstorage/config"
+	sourceConfig "github.com/conduitio/conduit-connector-google-cloudstorage/source/config"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+const (
+	projectID = "projectID"
+)
+
+func NewGCSClient(cfg map[string]string) (*storage.Client, error) {
```

Comment:
I believe it's better to have all the functions used only in tests unexported.

Reply:
These are the GCS util functions used across the both source integration tests and also the acceptance tests.
Here these tests are defined in different packages so i have exported them.

All the other functions specific to integration tests are kept unexported.
prepareIntegrationTest, addObjectsToTheBucket, readWithTimeout, readAndAssert ...

But still it can be possible if integration and acceptance tests are under one single package. Let me know should i go with this change.
Currently the source integration tests are in source package and acceptance tests are in connector package.

Reply:
IMO, that would be a good change, but if it will take time for any reason, we can open an issue for that and refactor later.

Reply:
OK will open an issue for that and refactor later (issue: #22 )